internal/store/serverservice/serverservice.go

```diff
@@ -222,69 +223,92 @@ func (r *Store) AssetUpdate(ctx context.Context, asset *model.Asset) error {
 				"hr": hr,
 			}).Warn("server service server query returned nil object")
 
-		return errors.Wrap(ErrServerServiceQuery, "got nil Server object")
+		return errors.Wrap(model.ErrInventoryQuery, "got nil Server object")
 	}
 
-	// create/update inventory.
-	if errInventory := r.createUpdateInventory(ctx, asset, server); errInventory != nil {
+	// publish server inventory
+	if errPublishInv := r.publishInventory(ctx, asset, server); errPublishInv != nil {
 		r.logger.WithFields(
 			logrus.Fields{
 				"id":  id,
 				"hr":  hr,
-				"err": errInventory.Error(),
-			}).Warn("inventory asset insert/update error")
+				"err": errPublishInv.Error(),
+			}).Warn("asset inventory insert/update error")
 
 		metricInventorized.With(prometheus.Labels{"status": "failed"}).Add(1)
-	} else {
-		// count devices with no errors
-		metricInventorized.With(prometheus.Labels{"status": "success"}).Add(1)
+
+		return errors.Wrap(model.ErrInventoryQuery, errPublishInv.Error())
 	}
 
-	// Don't publish bios config if there's no data
-	if len(asset.BiosConfig) != 0 {
-		err = r.createUpdateServerBIOSConfiguration(ctx, server.UUID, asset.BiosConfig)
-		if err != nil {
-			metricBiosCfgCollected.With(prometheus.Labels{"status": "failed"}).Add(1)
+	// count devices with no errors
+	metricInventorized.With(prometheus.Labels{"status": "success"}).Add(1)
 
-			r.logger.WithFields(
-				logrus.Fields{
-					"id":  server.UUID.String(),
-					"err": err,
-				}).Warn("error in server bios configuration versioned attribute update")
-		}
+	if errPublishBiosCfg := r.publishBiosConfig(ctx, asset, server); errPublishBiosCfg != nil {
+		r.logger.WithFields(
+			logrus.Fields{
+				"id":  server.UUID.String(),
+				"err": errPublishBiosCfg,
+			}).Warn("asset bios configuration insert/update error")
 
-		metricInventorized.With(prometheus.Labels{"status": "success"}).Add(1)
+		metricBiosCfgCollected.With(prometheus.Labels{"status": "failed"}).Add(1)
+
+		return errors.Wrap(model.ErrInventoryQuery, errPublishBiosCfg.Error())
 	}
 
+	metricBiosCfgCollected.With(prometheus.Labels{"status": "success"}).Add(1)
+
 	return nil
 }
 
-func (r *Store) createUpdateInventory(ctx context.Context, device *model.Asset, server *serverserviceapi.Server) error {
+func (r *Store) publishBiosConfig(ctx context.Context, asset *model.Asset, server *serverserviceapi.Server) error {
+	// Don't publish bios config if there's no bios config data made avai
+	if len(asset.BiosConfig) == 0 {
+		errNoBiosConfig := errors.New("no BIOS configuration collected")
```

Comment:
If there was an error collecting inventory, should we have already returned and bypassed this? Do we need this check?

Reply:
BIOS data is collected separately from inventory data and so it has to be published regardless of an inventory collection failure.